packages/nuxt/src/app/plugins/chunk-reload.client.ts

```diff
@@ -0,0 +1,23 @@
+import { defineNuxtPlugin } from '#app/nuxt'
+import { useRouter } from '#app/composables/router'
+
+export default defineNuxtPlugin((nuxtApp) => {
+  const router = useRouter()
+
+  let chunkError: TypeError | null = null
+  nuxtApp.hook('app:chunkError', ({ error }) => { chunkError = error })
+
+  router.onError((error, to) => {
+    if (chunkError !== error) { return }
+
+    let handledPath: Record<string, any> = {}
+    try {
+      handledPath = JSON.parse(localStorage.getItem('n:chunk:reload') ?? '{}')
+    } catch {}
+
+    if (handledPath?.path !== to.fullPath || handledPath?.expires < Date.now()) {
+      localStorage.setItem('n:chunk:reload', JSON.stringify({ path: to.fullPath, expires: Date.now() + 10000 }))
+      window.location.href = to.fullPath
```

Comment:
Can you please double-check if it is double URLs are not remaining in the history API?

Reply:
Check MDN docs: https://developer.mozilla.org/en-US/docs/Web/API/Location/href

Assigning makes a navigation (vs replacing as reload)

Reply:
But it's _meant_ to be a navigation.